### PR TITLE
fix: Initial Load Flickering.

### PR DESCRIPTION
**Pull request by amassarwi**
This improves the initial load flickering, before this the paywall message was displayed for some time and then
when the paywall info received from backend the message was gone.

This commit changes this behavior to hide the message by default and only after receiving from the backend that user
hasn't paid the message would show up.

Closes #2395.

### CODE REVIEW

File: src/actions/api.js [hunk 1]
```diff
@@ -95,14 +93,13 @@ export const onPollUserPayment = () => (dispatch, getState) => {
...
-      if (verified) {
-        dispatch(
-          act.UPDATE_USER_PAYWALL_STATUS({
-            status: PAYWALL_STATUS_PAID,
-            userid
-          })
-        );
-      } else {
+      dispatch(
+        act.UPDATE_USER_PAYWALL_STATUS({
```
Comment (amassarwi) [inline src/actions/api.js:97]: we rely on this action to determine is the user paid or not

File: src/components/PaywallMessage.jsx [hunk 2]
```diff
@@ -17,7 +17,8 @@ const PaywallMessage = ({ wrapper, ...props }) => {
       title: "Complete your registration",
       onClose: handleCloseModal
     });
-  const showMessage = !!currentUserEmail && !isPaid;
+  const showMessage = !!currentUserEmail && isPaid === false && isPaid !== null;
```
Comment (amassarwi) [inline src/components/PaywallMessage.jsx:20]: If `isPaid` is null we don't want to show the message as this means the isPaid info not available yet.

File: src/containers/Loader/Loader.js [hunk 3]
```diff
@@ -29,16 +28,13 @@ const Loader = ({ children }) => {
       try {
         const apiInfo = await onRequestApiInfo(false);
         setApiInfo(apiInfo);
-        if (apiInfo.activeusersession) {
```
Comment (amassarwi) [inline src/containers/Loader/Loader.js:32]: we have the same check already in `onRequestApiInfo`.

File: src/reducers/models/paywall.js [hunk 4]
```diff
@@ -11,18 +11,11 @@ const paywall = (state = DEFAULT_STATE, action) =>
     ? state
     : (
         {
-          [act.RECEIVE_ME || act.RECEIVE_LOGIN]: () => {
-            const { userid, paywalladdress } = action.payload;
-
-            return update(["byUserID", userid], (paywall) => ({
-              ...paywall,
-              isPaid: paywalladdress === ""
```
Comment (amassarwi) [inline src/reducers/models/paywall.js:19]: now we update `isPaid` only when on `UPDATE_USER_PAYWALL_STATUS` action, as the backend returns explicit 
`haspaid` field.

Comment (amassarwi) [reply src/reducers/models/paywall.js:19]: this was an issue for paid users; `isPaid: paywalladdress === ""` => resulted in `isPaid = false` this why the paywall 
message was visible for few moments until `UPDATE_USER_PAYWALL_STATUS` was dispatched with status paid which switched `isPaid` to true and only then the paywall message disappeared.

**Review by tiagoalvesdulce — COMMENTED** [review]
Please fix your PR description, it should close the issue https://github.com/decred/politeiagui/issues/2371.

I'm still seeing some ugly flickering in the UI contents. See:

https://user-images.githubusercontent.com/13955303/118498550-87b91680-b6fc-11eb-8d03-83f254a94812.mp4

I think the initial flicker reported by luke on https://github.com/decred/politeiagui/issues/2371 is fixed so you can choose to create new issues to deal with the issues I mentioned above.

**Comment by amassarwi** [comment]
@tiagoalvesdulce Thanks for the screen record, I would like to fix that in this PR, the paywall shouldn't flicker like that, 
i didn't see that while testing tho, will try now using slower network using the dev tools.

**Comment by tiagoalvesdulce** [comment]
OK, I'll create an issue.

**Review by tiagoalvesdulce — APPROVED** [review]

**Pull request merged**